Common play skill tests

forslund opened the pull request:
## Description
This adds unittests for the common play framework along with the semi-replated audioservice interface. Resolves #2332 

There are a couple of minor fixes and improvements
- track_info now uses the wait_for_response mechanic
- the queue now accepts tracks with mime info.
- A bunch of docstrings are improved

## How to test
Ensure that the travis tests passes.

## Contributor license agreement signed?
CLA [ Yes]

chrisveilleux commented on test/unittests/skills/test_audioservice.py:
```diff
@@ -0,0 +1,162 @@
+        audioservice = AudioService(bus)
+
+        # Test proper uri
+        audioservice.queue('file:///hello_nasty.mp3')
```
lots of duplicate code here. consider a function where you pass an argument for the queue and one for the actual value in the second assert statement

chrisveilleux commented on test/unittests/skills/test_common_play_skill.py:
```diff
@@ -0,0 +1,140 @@
+class TestCommonPlay(unittest.TestCase):
+    def test_lifecycle(self):
+        skill = CPSTest()
+        bus = mock.Mock(name='bus')
```
Use a setup method here that executes these two lines of code.  Make skill and bus instance attributes

forslund commented on test/unittests/skills/test_audioservice.py:
```diff
@@ -0,0 +1,162 @@
+        bus = mock.Mock(name='bus')
+        audioservice = AudioService(bus)
+
+        # Test proper uri
```
No, the it's just variations of input so I don't think it should be separate tests. If each check required a bunch of extra setups separating the tests would be called for but not when it's as simple as this.

forslund commented on test/unittests/skills/test_audioservice.py:
```diff
@@ -0,0 +1,162 @@
+
+class TestAudioService(unittest.TestCase):
+    def test_lifecycle(self):
+        bus = mock.Mock(name='bus')
```
I personally want the tests to be as easy to follow as possible without jumping around to see exactly how the setup was done. If this requires some duplication of setup code that's fine by me.

forslund replied:
Especially since one of the tests include the init I don't want it to be in the setup

forslund commented:
I refactored out the setup code in the test_common_play_skill since after looking at it again it was slightly complex and it made sense to move it out. I also split up the test of match play query and moved it to a separate testcase class to make sure the test was grouped logically.

The test_audioservice setup was so trivial so I left it as is, moving it to class members just makes it more fragmented and difficult to read.

forslund commented:
Did some more refactoring of the tests, the multi tests have been split out into separate classes to make sure it's shown that they're related.

Some of the code duplication has been removed while some was kept to keep the logic of the test clear.
